### PR TITLE
[T3A3][T16-A2]ChingHuiQi

### PR DESCRIPTION
**Pull request by HQching**

### CODE REVIEW

**Comment by HQching** [comment]
Some code referred from #6 (usage of split method)
Ready for review

**Comment by HQching** [comment]
Ready for review (Sorry, please ignore the first commit)

File: src/seedu/addressbook/data/person/Address.java [hunk 1]
```diff
@@ -21,36 +138,51 @@
...
+     * @throws IllegalValueException if any of the above have invalid format
+     */
+    private void setAddressParams(boolean isPrivate, String addressParams) throws IllegalValueException {
+        this.isPrivate = isPrivate;
+        final Matcher matcher = ADDRESS_VALIDATION_REGEX.matcher(addressParams);
+        matcher.matches();
+        block = new Block(matcher.group("BLOCK"));
+        street = new Street(matcher.group("STREET"));
+        unit = new Unit(matcher.group("UNIT"));
+        postalCode = new PostalCode(matcher.group("POSTALCODE"));
```
Comment (brandonyeoxg) [inline src/seedu/addressbook/data/person/Address.java]: Good job using named capturing groups for readability =).

File: test/java/seedu/addressbook/parser/ParserTest.java [hunk 2]
```diff
@@ -72,7 +72,7 @@ public void parse_exitCommand_parsedCorrectly() {
     }
 
     /*
-     * Tests for ingle index argument commands ===============================================================
+     * Tests for single index argument commands ===============================================================
```
Comment (brandonyeoxg) [inline test/java/seedu/addressbook/parser/ParserTest.java]: Glad you updated this comment =).

File: src/seedu/addressbook/data/person/Address.java [hunk 3]
```diff
@@ -21,36 +138,51 @@
...
+
+    /**
+     * Splits address into block, street, unit and postal code
+     * 
+     * @throws IllegalValueException if any of the above have invalid format
+     */
+    private void setAddressParams(boolean isPrivate, String addressParams) throws IllegalValueException {
+        this.isPrivate = isPrivate;
+        final Matcher matcher = ADDRESS_VALIDATION_REGEX.matcher(addressParams);
+        matcher.matches();
```
Comment (brandonyeoxg) [inline src/seedu/addressbook/data/person/Address.java]: Be careful when using Regex.
In this case what happens when `matcher.matches()` returns a false?
What happens if the address provided does not fit your Regex?

**Pull request closed without merging**